test/parallel/test-crypto-subtle-zero-length.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const crypto = require('crypto').webcrypto;
+
+crypto.subtle.importKey(
+  'raw',
+  new Uint8Array(32),
+  { name: 'AES-GCM' },
+  false,
+  [ 'encrypt', 'decrypt' ])
+  .then((k) => {
+    assert(k instanceof crypto.CryptoKey);
+    return crypto.subtle.encrypt({
+      name: 'AES-GCM',
+      iv: new Uint8Array(12),
+    }, k, new Uint8Array(0)).then((e) => {
+      assert(e instanceof ArrayBuffer);
+      assert.deepStrictEqual(
+        Buffer.from(e),
+        Buffer.from([
+          0x53, 0x0f, 0x8a, 0xfb, 0xc7, 0x45, 0x36, 0xb9,
+          0xa9, 0x63, 0xb4, 0xf1, 0xc4, 0xcb, 0x73, 0x8b ]));
+      return crypto.subtle.decrypt(
+        {
+          name: 'AES-GCM',
+          iv: new Uint8Array(12)
+        },
+        k,
+        e);
+    });
+  })
+  .then((v) => {
+    assert(v instanceof ArrayBuffer);
+    assert.strictEqual(v.byteLength, 0);
+  });
```

Comment:
Since this is quite difficult to read and there's multiple `then(...)` chains, it is probably more readable and simpler to understand to use an `async` function here.

Reply:
I've updated